packages/graphql-hooks/src/GraphQLClient.ts

```diff
@@ -407,6 +414,31 @@ class GraphQLClient {
       return this.subscriptionClient.request(operationPayload)
     }
   }
+
+  invalidateQuery(query: string): void {
+    const cacheKey = this.getCacheKey({ query })
+    if (this.cache && cacheKey) {
+      this.removeCache(cacheKey)
+      this.request({ query }).then(result =>
+        this.mutationsEmitter.emit(Events.DATA_INVALIDATED, result)
+      )
+    }
+  }
+
+  setQueryData(query: string, updater: (oldState?: any) => any): void {
+    const cacheKey = this.getCacheKey({ query })
+    if (this.cache && cacheKey) {
+      const oldState: any = this.cache.get(cacheKey)
+      const newState = {
+        ...oldState,
+        data: {
+          ...updater(oldState.data || null)
```

Comment:
as we are already creating a new object here, is this spread necessary?
can we do just
```
data: updater(...)
```
What do you think?

Reply:
made in https://github.com/nearform/graphql-hooks/pull/928/commits/4ea3bd85457943db1b1032920ce4d5d03901d171